ppb/sprites.py

```diff
@@ -19,6 +26,113 @@
 side_attribute_error_message = error_message.format
 
 
+class BaseSprite(EventMixin):
+    """
+    The base Sprite class. All sprites should inherit from this (directly or
+    indirectly).
+
+    The things that define a BaseSprite:
+
+    * The __event__ protocol (see ppb.eventlib.EventMixin)
+    * A position vector
+    * A layer
+
+    BaseSprite provides an __init__ method that sets attributes based on kwargs
+    to make rapid prototyping easier.
+    """
+    #: (:py:class:`ppb.Vector`): Location of the sprite
+    position: Vector = Vector(0, 0)
+    #: The layer a sprite exists on.
+    layer: int = 0
+
+    def __init__(self, **kwargs):
+        super().__init__()
+
+        self.position = Vector(self.position)
+
+        # Initialize things
+        for k, v in kwargs.items():
+            # Abbreviations
+            if k == 'pos':
+                k = 'position'
+            # Castings
+            if k == 'position':
+                v = Vector(v)
+            setattr(self, k, v)
+
+
+class RenderableMixin:
+    """
+    A class implementing the API expected by ppb.systems.renderer.Renderer.
+
+    You should include RenderableMixin after BaseSprite in your parent class
```

Comment:
`RenderableMixin` should come before `BaseSprite`, assuming that `BaseSprite` is the conceptual parent to `RenderableMixin`.